pkg/providers/utils/ingress_class.go

```diff
@@ -0,0 +1,31 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package utils
+
+import "github.com/apache/apisix-ingress-controller/pkg/config"
+
+func MatchCRDsIngressClass(resourceIngressClassName string, configIngressClass string) bool {
```

Comment:
Will the resourceIngressClassName field be empty (nil)?  This field does not exist for legacy resources.

Reply:
If it is empty, it will return false. The old resource will be handed over to the caller for judgment. This method is only used to match the ingressClass